scenes/skillchecks/chalk/ChalkSkillCheck.gd

```diff
@@ -0,0 +1,260 @@
+extends "res://scenes/skillchecks/SkillCheck.gd"
+
+const EASY = 1
+const MED = 2
+const HARD = 3
+
+const CircleNode = preload("./CircleNode.tscn")
+const score_text = "Score: %d\nLives: %d"
+var rng = RandomNumberGenerator.new()
+
+var difficulty = MED
+var path = []
+var generated_path = []
+var pressed = false
+var positions
+var failed = false
+var score = 0
+var lives = 3
+var flash_delay = 0.4
+var delay_increment = 0.9
+
+var chalk_start_point = Vector2(40, 100)
+
+var path_length = {
+	EASY: 5,
+	MED: 6,
+	HARD: 8
+}
+
+var nodes = {
+	EASY: [0, 1, 2, 5, 6, 7, 10, 11, 12],
+	MED: [0, 1, 2, 3, 5, 6, 7, 8, 10, 11, 12, 13, 15, 16, 17, 18],
+	HARD: [0, 1, 2, 3, 5, 6, 7, 8, 10, 11, 12, 13, 15, 16, 17, 18]
+#	HARD: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12,
+#			13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
+}
+
+var node_ids = {}
+
+const neighbors = {
+    0: [1, 5, 6],
+    1: [0, 2, 5, 6, 7],
+    2: [1, 3, 6, 7, 8],
+    3: [2, 4, 7, 8, 9],
+    4: [3, 8, 9],
+    5: [0, 1, 6, 10, 11],
+    6: [0, 1, 2, 5, 7, 10, 11, 12],
+    7: [1, 2, 3, 6, 8, 11, 12, 13],
+    8: [2, 3, 4, 7, 9, 12, 13, 14],
+    9: [3, 4, 8, 13, 14],
+    10: [5, 6, 11, 15, 16],
+    11: [5, 6, 7, 10, 12, 15, 16, 17],
+    12: [6, 7, 8, 11, 13, 16, 17, 18],
+    13: [7, 8, 9, 12, 14, 17, 18, 19],
+    14: [8, 9, 13, 18, 19],
+    15: [10, 11, 16, 20, 21],
+    16: [10, 11, 12, 15, 17, 20, 21, 22],
+    17: [11, 12, 13, 16, 18, 21, 22, 23],
+    18: [12, 13, 14, 17, 19, 22, 23, 24],
+    19: [13, 14, 18, 23, 24],
+    20: [15, 16, 21],
+    21: [15, 16, 17, 20, 22],
+    22: [16, 17, 18, 21, 23],
+    23: [17, 18, 19, 22, 24],
+    24: [18, 19, 23]
+}
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	$UI/ChalkIcon.position = chalk_start_point
+	print(ProjectSettings.get_setting("display/window/size/height"))
+
+	randomize()
```

Comment:
We could move this later to `GlobalVariables` or something since `randomize` only needs to be run once for the entire game.